Signal/src/view controllers/CodeVerificationViewController.m

```diff
@@ -474,9 +458,17 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
 }
 
 - (void)setVerificationCodeAndTryToVerify:(NSString *)verificationCode {
-    NSCharacterSet *validCharacterSet = [NSCharacterSet decimalDigitCharacterSet];
-    self.challengeTextField.text = [self stringByFilteringString:verificationCode
-                                                withCharacterSet:validCharacterSet];
+    NSString *rawNewText = verificationCode.digitsOnly;
+    NSString *formattedNewText = (rawNewText.length <= 3
+                                  ? rawNewText
+                                  : [[[rawNewText substringToIndex:3]
+                                      stringByAppendingString:@"-"]
+                                     stringByAppendingString:[rawNewText substringFromIndex:3]]);
+    self.challengeTextField.text = formattedNewText;
+    // Move the cursor after the newly inserted text.
+    UITextPosition *newPosition = [self.challengeTextField endOfDocument];
+    self.challengeTextField.selectedTextRange = [self.challengeTextField textRangeFromPosition:newPosition
```

Comment:
Not broken, but the code would show up as "123456"; it should show up as "123-456".